Amazon.IonHashDotnet.Tests/ReaderCompare.cs

```diff
@@ -183,15 +183,25 @@ private static void CompareScalars(IonType t, bool isNull, IIonReader it1, IIonR
                         {
                             BigDecimal bigDec1 = it1.DecimalValue();
                             BigDecimal bigDec2 = it2.DecimalValue();
-                            AssertPreciselyEquals(bigDec1, bigDec2);
 
-                            decimal dec1 = bigDec1.ToDecimal();
-                            decimal dec2 = bigDec2.ToDecimal();
-                            Assert.AreEqual(dec1, dec2);
+                            try
+                            {
+                                decimal dec1 = bigDec1.ToDecimal();
+                                decimal dec2 = bigDec2.ToDecimal();
+                                Assert.AreEqual(dec1, dec2);
 
-                            Assert.AreEqual(decimal.ToDouble(dec1), decimal.ToDouble(dec2));
-                            Assert.AreEqual(decimal.ToInt32(dec1), decimal.ToInt32(dec2));
-                            Assert.AreEqual(decimal.ToInt64(dec1), decimal.ToInt64(dec2));
+                                Assert.AreEqual(decimal.ToDouble(dec1), decimal.ToDouble(dec2));
+                                Assert.AreEqual(decimal.ToInt32(dec1), decimal.ToInt32(dec2));
+                                Assert.AreEqual(decimal.ToInt64(dec1), decimal.ToInt64(dec2));
+                            }
+                            catch (OverflowException)
+                            {
+                                // Certain Ion Decimals in test file exceed C# decmial range. Just AssertPreciselyEquals.
+                            }
+                            finally
+                            {
+                                AssertPreciselyEquals(bigDec1, bigDec2);
+                            }
```

Comment:
Help me understand why this assertion is moving from its previous location into a finally block?

Reply:
My thought was that if it's moved here, then the test will be run even if the tests in the try block fail, to help narrow down the source of failure, or at least tell the user that one test passed and the other test fail.

Looking back, it doesn't really matter, so I'll undo this.